### PR TITLE
Update SetupCrossTenantRelationshipForTargetTenant.ps1

### PR DESCRIPTION
**Pull request by ropow**
Govt. cloud update

### CODE REVIEW

File: SetupCrossTenantRelationshipForTargetTenant.ps1 [hunk 1]
```diff
@@ -685,7 +685,28 @@ function Run-ExchangeSetupForTargetTenant([string]$targetTenant, [string]$resour
...
+                                        -ApplicationId $appId `
+                                        -AppSecretKeyVaultUrl $appSecretKeyVaultUrl `
+                                        -ExchangeRemoteMove:$true
+    }
+    else
+    {
+        $global:MigrationEndpoint = New-MigrationEndpoint `
+                                        -Name $orgRelName `
+                                        -RemoteTenant $resourceTenantDomain `
+                                        -RemoteServer "outlook.office.us" `
```
Comment (shrewdeer) [inline SetupCrossTenantRelationshipForTargetTenant.ps1:703]: should this be outlook.office.com?

**Comment by ropow** [comment]
Not when the -Government switch is passed, it's .com in the elseif

**Comment by psykokwak4** [comment]
> Not when the -Government switch is passed, it's .com in the elseif From: shrewdeer ***@***.***> Sent: Monday, March 15, 2021 11:38 AM To: microsoft/cross-tenant ***@***.***> Cc: Roman Powell ***@***.***>; Author ***@***.***> Subject: Re: [microsoft/cross-tenant] Update SetupCrossTenantRelationshipForTargetTenant.ps1 (#38) @shrewdeer commented on this pull request.
> ________________________________ In SetupCrossTenantRelationshipForTargetTenant.ps1<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fcross-tenant%2Fpull%2F38%23discussion_r594592175&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069460022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ELU%2B6MOe2bSU756iQcewM764Wxf7%2BlD2nOYDWmA37%2BY%3D&reserved=0>:
> + if (-not $MigrationEndpointMaxConcurrentMigrations -and $Government -eq $true)
> + { + $global:MigrationEndpoint = New-MigrationEndpoint ` + -Name $orgRelName ` + -RemoteTenant $resourceTenantDomain ` + -RemoteServer "outlook.office.us" ` + -ApplicationId $appId ` + -AppSecretKeyVaultUrl $appSecretKeyVaultUrl ` + -ExchangeRemoteMove:$true + } + else + { + $global:MigrationEndpoint = New-MigrationEndpoint ` + -Name $orgRelName ` + -RemoteTenant $resourceTenantDomain ` + -RemoteServer "outlook.office.us" ` should this be outlook.office.com? - You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fcross-tenant%2Fpull%2F38%23pullrequestreview-612528241&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069460022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wZBxy%2BAKGd%2F6LsAU44bTizxe9L5U4f5Tu0JPgu%2BiEdg%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPUHY6C2GN4R62DOG5LTHWDTDZH2DANCNFSM4ZHB4EKQ&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069470012%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ex%2FJMVrYeeBwac3tRW%2FbAys22wlMoojRbQ%2FYxHEJkZE%3D&reserved=0>.

isn't `else` block on line 698 capturing all "else" conditions which renders `elseif` block on line 709 useless?

image

**Comment by ropow** [comment]
> > Not when the -Government switch is passed, it's .com in the elseif From: shrewdeer _**@**_._**> Sent: Monday, March 15, 2021 11:38 AM To: microsoft/cross-tenant _**@**_.**_> Cc: Roman Powell _**@**_._**>; Author _**@**_.**_> Subject: Re: [microsoft/cross-tenant] Update SetupCrossTenantRelationshipForTargetTenant.ps1 (#38) @shrewdeer commented on this pull request.
> > ________________________________ In SetupCrossTenantRelationshipForTargetTenant.ps1https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fcross-tenant%2Fpull%2F38%23discussion_r594592175&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069460022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ELU%2B6MOe2bSU756iQcewM764Wxf7%2BlD2nOYDWmA37%2BY%3D&reserved=0:
> > 
> > * if (-not $MigrationEndpointMaxConcurrentMigrations -and $Government -eq $true)
> > * { + $global:MigrationEndpoint = New-MigrationEndpoint `+ -Name $orgRelName` + -RemoteTenant $resourceTenantDomain `+ -RemoteServer "outlook.office.us"` + -ApplicationId $appId `+ -AppSecretKeyVaultUrl $appSecretKeyVaultUrl` + -ExchangeRemoteMove:$true + } + else + { + $global:MigrationEndpoint = New-MigrationEndpoint `+ -Name $orgRelName` + -RemoteTenant $resourceTenantDomain `+ -RemoteServer "outlook.office.us"` should this be outlook.office.com? - You are receiving this because you authored the thread. Reply to this email directly, view it on GitHubhttps://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fcross-tenant%2Fpull%2F38%23pullrequestreview-612528241&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069460022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wZBxy%2BAKGd%2F6LsAU44bTizxe9L5U4f5Tu0JPgu%2BiEdg%3D&reserved=0, or unsubscribehttps://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPUHY6C2GN4R62DOG5LTHWDTDZH2DANCNFSM4ZHB4EKQ&data=04%7C01%7CRoman.Powell%40microsoft.com%7C9ea6e421837a4545812c08d8e7e184b4%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637514303069470012%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ex%2FJMVrYeeBwac3tRW%2FbAys22wlMoojRbQ%2FYxHEJkZE%3D&reserved=0.
> 
> isn't `else` block on line 698 capturing all "else" conditions which renders `elseif` block on line 709 useless?
> 
> image

Yeah, I was a bit confused on how to go about doing this. New iteration.

**Comment by psykokwak4** [comment]
just learned the below marvelous powershell trick before, do you think if it can be used to remove duplicate New-MigrationEndpoint invocations?

```
PS C:\> $params=@{}
PS C:\> $params['Name']='haha'
PS C:\> $params['RemoteServer']='haha.com'
PS C:\> $params['ExchangeRemoteMove']=$true
PS C:\> $params['SkipVerification']=$true
PS C:\> $params['WhatIf']=$true
PS C:\> & New-MigrationEndpoint @params
What if: Do you want to create the migration endpoint "haha"?
```

**Review by psykokwak4 — APPROVED** [review]

**Review by shrewdeer — APPROVED** [review]
Thanks.

**Review by shankaraarun — APPROVED** [review]

**Pull request merged**